### ISSUE TITLE
Use a Cell Type cluster and not a name of cell to run Palantir

### ISSUE THREAD

**Issue by emanuelavilla** [dpeerlab/Palantir#83]
Hello!

I am opening an issue because I have a question about the choice to use a cell ID and not a cell type cluster in the command run_palantir. I cannot completely understand the reason behind this, since using a cell type could be easier to be identified as a trajectory start. 

Best, 
Emanuela

**Comment by ManuSetty** [comment]
The distances are all computed using the start cell and hence we request the specification of a cell and not cluster. If you know the start cluster, you can choose the extreme diffusion component cell as the start cell

**Comment by wangjiawen2013** [comment]
I agree with @emanuelavilla , monocle supports using cluster name as the start point. However, If I use the start cell, how to choose the extreme diffusion component cell ? What is the extreme diffusion component ?

**Comment by katosh** [comment]
We recently implemented a small utility to find a start cell based on it being the extrem end of a diffusion component. The rational is, that such a cell is the extrem end of the biological process that causes major variability in the dataset, e.g., the differentiation process. The command would be:
```python
start_cell = palantir.utils.early_cell(anndata, celltype, celltype_column_in_obs)
```
You can find documentation on this call [here](https://palantir.readthedocs.io/en/latest/utils.html#palantir.utils.early_cell).

Equivalently, if you want to find multiple terminal cell-states you can use
```python
terminals = palantir.utils.find_terminal_states(anndata, celltype, celltype_column_in_obs)
```

Note, that neither of these two calls are guaranteed to find a cell. In the case that there is no cell in the cell type that sits at an extreme of any of the diffusion components, then the calls will fail. So, to guarantee successful execution, e.g., when using this in a pipeline, we implemented a slow fallback algorithm. It recomputes pseudo time starting from a random cell outside of the respective cell type and selects the "oldest" cell of the cell type as an extrem. You can enable the fallback algorithm by passing an integer as the `fallback_seed` to the calls.

Please make sure the identified cells are appropriate. You can look at them on the umap, e.g., with
```python
palantir.plot.highlight_cells_on_umap(ad, {"start cell": start_cell})
plt.show()
palantir.plot.highlight_cells_on_umap(ad, terminals)
plt.show()
```
